src/main/java/com/team4/testingsystem/converters/TestConverter.java

```diff
@@ -0,0 +1,62 @@
+package com.team4.testingsystem.converters;
+
+import com.team4.testingsystem.dto.QuestionDTO;
+import com.team4.testingsystem.dto.TestDTO;
+import com.team4.testingsystem.entities.ContentFile;
+import com.team4.testingsystem.entities.Question;
+import com.team4.testingsystem.entities.Test;
+import com.team4.testingsystem.enums.Modules;
+import com.team4.testingsystem.services.ContentFilesService;
+import com.team4.testingsystem.services.QuestionService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Component
+public class TestConverter {
+
+    private final QuestionService questionService;
+    private final ContentFilesService contentFilesService;
+
+    @Autowired
+    public TestConverter(QuestionService questionService,
+                         ContentFilesService contentFilesService) {
+        this.questionService = questionService;
+        this.contentFilesService = contentFilesService;
+    }
+
+    public TestDTO convertToDTO(Test test) {
+        TestDTO testDTO = new TestDTO(test);
+        setQuestions(testDTO, test.getId());
+        testDTO.setContentFile(appendContentFile(test.getId()).getUrl());
```

Comment:
we already set questions to dto here. Can we get its id here, instead of passing testId and get a question from DB once again?